crates/routes/src/images.rs

```diff
@@ -123,6 +128,19 @@ async fn full_res(
   client: web::Data<ClientWithMiddleware>,
   context: web::Data<LemmyContext>,
 ) -> Result<HttpResponse, Error> {
+  // block access to images if instance is private and unauthorized, public
+  let site = blocking(context.pool(), Site::read_local_site).await?;
+  // The site might not be set up yet
+  if let Ok(site) = site {
+    if site.private_instance {
+      let jwt = req
+      .cookie("jwt")
+      .expect("No auth header for picture access");
+      if get_local_user_view_from_jwt(jwt.value(), context.pool(), context.secret()).await.is_err() {
+        return Ok(HttpResponse::Unauthorized().finish());
```

Comment:
I think its better to return the `Error(error::ErrorUnauthorized)`

Reply:
I think in both cases we should return a HttpResponse. I'm not sure how to do that honestly. 

Is replacing expect() with .map_err(error::ErrorUnauthorized) enough above?

Reply:
I spose that's fine, althoug that actix errors are also HttpResponses.

Reply:
@dessalines I'm getting errors about types not being correct. I'm really unsure how to do that, and need help on those - sorry not yet familiar with those rust libs.

Reply:
There are other examples in this file. If you run `cargo check`, it should tell you the exact error.